crypto/x509/x_pubkey.c

```diff
@@ -977,20 +977,25 @@ int ossl_i2d_X448_PUBKEY(const ECX_KEY *a, unsigned char **pp)
 
 #endif
 
+void X509_PUBKEY_set0_public_key(X509_PUBKEY *pub,
+                                 unsigned char *penc, int penclen)
+{
+    OPENSSL_free(pub->public_key->data);
+    pub->public_key->data = penc;
+    pub->public_key->length = penclen;
+    /* Set number of unused bits to zero */
+    pub->public_key->flags &= ~(ASN1_STRING_FLAG_BITS_LEFT | 0x07);
```

Comment:
This bit of code is duped in quite a few places now. is it worth making it a function?

Reply:
Good point.
Added an internal function `asn1_string_set_bits_left()` that is now used wherever possible.

Reply:
Doing so, I found that there is much more code overlap on ASN1 bits fiddling in libcrypto, but don't wanna stretch this PR.